### DISCUSSION TITLE
Example screen pytest fixture is likely too strict about logs

### DISCUSSION THREAD

**Discussion by jeffective** [zauberzeug/nicegui#2829] [Q&A]
### Question

the example screen fixture calls pytest.fail if there are any logs. This should probably be left to the user to configure.

this is the fixture I wrote, I didnt know what to do with downloads and I dont have any so i just deleted it.

conftest.py:
```python
from collections.abc import Generator

import pytest
from nicegui.testing import Screen
from nicegui.testing.conftest import (
    capabilities,  # noqa: F401
    chrome_options,  # noqa: F401
    driver,  # noqa: F401
    remove_all_screenshots,  # noqa: F401
    reset_globals,  # noqa: F401
)
from selenium import webdriver


@pytest.fixture
def screen(
    driver: webdriver.Chrome,  # noqa: F811
    request: pytest.FixtureRequest,
    caplog: pytest.LogCaptureFixture,
) -> Generator[Screen, None, None]:
    """Create a new Screen instance."""
    screen_ = Screen(driver, caplog)
    yield screen_
    if screen_.is_open:
        screen_.shot(request.node.name)
    screen_.stop_server()


```

**Comment by jeffective** [comment]
also , it might be best to only screenshot if a test fails?

**Comment by rodja** [comment]
> the example screen fixture calls pytest.fail if there are any logs. This should probably be left to the user to configure.

Yes, that could be a valuable addition. Would you like to create a pull request?

> I didnt know what to do with downloads and I dont have any so i just deleted it.

The default should support downloads in my opinion. But of course you can change that in your testing code.

> also , it might be best to only screenshot if a test fails?

Why? Is the screenshot taking to much time? Otherwise they serve as a documentation for a successful executed test.
